packages/salesforcedx-sobjects-faux-generator/test/integration/fauxGenerate.test.ts

```diff
@@ -0,0 +1,200 @@
+    invalidateProject(projectPath);
+    try {
+      result = await generator.generate(projectPath, SObjectCategory.CUSTOM);
+    } catch (e) {
+      expect(e).to.contain(FAILURE_CODE);
+      expect(e).to.contain(nls.localize('no_generate_if_not_in_project', ''));
+      return;
+    } finally {
+      restoreProject(projectPath);
+    }
+    expect.fail(result, 'undefined', 'generator should have thrown an error');
+  });
+
+  it('Should emit an error event on failure', async () => {
+    let result = '';
+    let errorCode = SUCCESS_CODE;
+    const generator = getGenerator();
+    emitter.addListener(LocalCommandExecution.ERROR_EVENT, (data: string) => {
+      errorCode = data;
+    });
+    invalidateProject(projectPath);
+    try {
+      result = await generator.generate(projectPath, SObjectCategory.CUSTOM);
+    } catch (e) {
+      expect(e).to.contain(FAILURE_CODE);
+    } finally {
+      restoreProject(projectPath);
+    }
+    expect(errorCode).to.equal(FAILURE_CODE);
+  });
+
+  it('Should emit message to stderr on failure', async () => {
+    let result = '';
+    let stderrInfo = '';
+    const generator = getGenerator();
+    emitter.addListener(LocalCommandExecution.STDERR_EVENT, (data: string) => {
+      stderrInfo = data;
+    });
+    invalidateProject(projectPath);
+    try {
+      result = await generator.generate(projectPath, SObjectCategory.CUSTOM);
+    } catch (e) {
+      expect(e).to.contain(FAILURE_CODE);
+    } finally {
+      restoreProject(projectPath);
+    }
+    expect(stderrInfo).to.contain(
+      nls.localize('no_generate_if_not_in_project', '')
+    );
+  });
+
+  it('Should emit an exit event with code success code 0 on success', async () => {
+    let result = '';
+    let exitCode = FAILURE_CODE;
+    const generator = getGenerator();
+    emitter.addListener(LocalCommandExecution.EXIT_EVENT, (data: string) => {
+      exitCode = data;
+    });
+    try {
+      result = await generator.generate(projectPath, SObjectCategory.CUSTOM);
+    } catch (e) {
+      expect.fail(e, 'undefined', 'generator should not have thrown an error');
+    }
+    expect(result).to.equal(SUCCESS_CODE);
+    expect(exitCode).to.equal(SUCCESS_CODE);
+  });
+
+  it('Should log the number of created faux classes on success', async () => {
+    const generator = getGenerator();
+    let stdoutInfo = '';
+    let result = '';
+    emitter.addListener(LocalCommandExecution.STDOUT_EVENT, (data: string) => {
+      stdoutInfo = data;
+    });
+    try {
+      result = await generator.generate(projectPath, SObjectCategory.CUSTOM);
+    } catch (e) {
+      expect.fail(e, 'undefined', 'generator should not have thrown an error');
+    }
+    expect(result).to.equal(SUCCESS_CODE);
+    expect(stdoutInfo).to.contain(
+      nls.localize('fetched_sobjects_length_text', 3, 'Custom')
+    );
+  });
+});
+
+// easy way to force the generator to throw an error
+const DUMMY_PROJECT_FILE = `zzz${SFDX_PROJECT_FILE}`;
```

Comment:
Don't do `zzz`. Either use fake, bogus, or some other adjective. It's very annoying to have to read a zzz, xyz, and not understand why someone named it that way. It's the same with CUSTOM_OBJECT_2, CUSTOM_OBJECT_3,

Reply:
As I noted before, CUSTOM_OBJECT_2, etc are because they are all the same and just there to deal with having multiple custom objects...    But I'll change zzz - I can understand that concern.